hutch_python/epics_arch.py

```diff
@@ -93,13 +116,138 @@ def create_arch_file(experiment, hutch=None, path=None, dry_run=False):
             file_path = path
         elif hutch:
             file_path = EPICS_ARCH_FILE_PATH.format(hutch.lower())
+        elif cds_items:
+            pull_cds_data(experiment, cds_items)
+            return
+        elif link:
+            update_file(exp_name=experiment, path=EPICS_ARCH_FILE_PATH.format(experiment[0:3]))
+            create_softlink(experiment)
+            return
         else:
             file_path = EPICS_ARCH_FILE_PATH.format(experiment[0:3])
-        create_file(exp_name=experiment, path=file_path)
+        update_file(exp_name=experiment, path=file_path)
     elif dry_run:
         print_dry_run(experiment)
 
 
+def pull_cds_data(exp, run):
+    logger.debug("in client")
+    pull_cds_items(exp, run)
+
+
+def create_softlink(experiment):
+    logger.debug("in softlink")
+    # remove the old soft link and add a new one (update), *THIS HAS NOT BEEN TESTED YET*
+    # this removes the softlink in the /cds/group/pcds/dist/pds/{}/misc/
+    # rm_result = subprocess.run(['unlink', EPICS_ARCH_FILE_PATH.format(experiment[0:3]) + 'epicsArch_' + experiment[0:3].upper() + '_exp_specific.txt'])
+
+    # This adds a new softlink in /cds/group/pcds/dist/pds/{}/misc/
```

Comment:
DOC: you should probably correct the comments here and remake them into a docstring